src/GeoJson/GeoJson.php

```diff
@@ -101,56 +84,61 @@ final public static function jsonUnserialize($json)
             case 'MultiPolygon':
             case 'Point':
             case 'Polygon':
-                if ( ! $json->offsetExists('coordinates')) {
+                if (! $json->offsetExists('coordinates')) {
                     throw UnserializationException::missingProperty($type, 'coordinates', 'array');
                 }
 
-                if ( ! is_array($json['coordinates'])) {
+                if (! is_array($json['coordinates'])) {
                     throw UnserializationException::invalidProperty($type, 'coordinates', $json['coordinates'], 'array');
                 }
 
                 $args[] = $json['coordinates'];
                 break;
 
             case 'Feature':
-                $geometry = isset($json['geometry']) ? $json['geometry'] : null;
-                $properties = isset($json['properties']) ? $json['properties'] : null;
+                $geometry = $json['geometry'] ?? null;
+                $properties = $json['properties'] ?? null;
+                $id = $json['id'] ?? null;
 
-                if (isset($geometry) && ! is_array($geometry) && ! is_object($geometry)) {
+                if ($geometry !== null && ! is_array($geometry) && ! is_object($geometry)) {
                     throw UnserializationException::invalidProperty($type, 'geometry', $geometry, 'array or object');
                 }
 
-                if (isset($properties) && ! is_array($properties) && ! is_object($properties)) {
+                if ($properties !== null && ! is_array($properties) && ! is_object($properties)) {
                     throw UnserializationException::invalidProperty($type, 'properties', $properties, 'array or object');
                 }
 
-                $args[] = isset($geometry) ? self::jsonUnserialize($geometry) : null;
-                $args[] = isset($properties) ? (array) $properties : null;
-                $args[] = isset($json['id']) ? $json['id'] : null;
+                if ($id !== null && ! is_int($id) && ! is_string($id)) {
+                    throw UnserializationException::invalidProperty($type, 'id', $id, 'int or string');
+                }
```

Comment:
```suggestion
                // TODO: Validate non-null $id as int or string in 2.0
```

Thanks for calling this out in the OP. I agree that this is a subtle BC break and better pushed to 2.0. Please revert this type-check and any corresponding test. I think it'd also make sense to replace this line with a TODO comment for the time being.

Reply:
Done! 🙌